public/app/features/alerting/unified/components/rules/Filter/RulesFilter.v2.tsx

```diff
@@ -1,226 +1,769 @@
+  };
+
+  const handleOnToggle = () => {
+    trackFilterButtonClick();
+    setIsPopupOpen(!isPopupOpen);
+  };
+
+  // Handle outside clicks to close the popup
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (isPopupOpen && popupRef.current && event.target instanceof Node && !popupRef.current.contains(event.target)) {
+        // Check if click is on a portal element (combobox dropdown)
+        if (event.target instanceof Element) {
+          const isPortalClick =
+            event.target.closest('[data-popper-placement]') || event.target.closest('[role="listbox"]');
+
+          if (!isPortalClick) {
+            setIsPopupOpen(false);
+          }
+        } else {
+          setIsPopupOpen(false);
         }
-        header={
-          <TabsBar hideBorder className={styles.fixTabsMargin}>
-            <Tab
-              active={activeTab === 'custom'}
-              icon="filter"
-              label={t('alerting.rules-filter.filter-options.label-custom-filter', 'Custom filter')}
-              onChangeTab={() => setActiveTab('custom')}
-            />
-            <Tab
-              active={activeTab === 'saved'}
-              icon="bookmark"
-              label={t('alerting.rules-filter.filter-options.label-saved-searches', 'Saved searches')}
-              onChangeTab={() => setActiveTab('saved')}
+      }
+    };
+
+    if (isPopupOpen) {
+      document.addEventListener('mousedown', handleClickOutside);
+    }
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [isPopupOpen]);
+
+  const filterButtonLabel = t('alerting.rules-filter.filter-options.aria-label-show-filters', 'Filter');
+  return (
+    <form onSubmit={handleSubmit(submitHandler)} onReset={() => {}}>
+      <Stack direction="column" gap={1}>
+        <Label htmlFor="rulesSearchInput">
+          <Stack gap={0.5} alignItems="center">
+            <span>
+              <Trans i18nKey="alerting.rules-filter.search">Search</Trans>
+            </span>
+            <PopupCard content={<SearchQueryHelp />}>
+              <Icon
+                name="info-circle"
+                size="sm"
+                tabIndex={0}
+                title={t('alerting.rules-filter.title-search-help', 'Search help')}
+              />
+            </PopupCard>
+          </Stack>
+        </Label>
+        <Stack direction="row" alignItems="center" gap={1}>
+          <div style={{ flex: 1 }}>
+            <FilterInput
+              id="rulesSearchInput"
+              data-testid="search-query-input"
+              placeholder={t(
+                'alerting.rules-filter.filter-options.placeholder-search-input',
+                'Search by name or enter filter query...'
+              )}
+              name="searchQuery"
+              onChange={(string) => setValue('query', string)}
+              onBlur={() => {
+                const currentQuery = watch('query');
```

Comment:
`getValues` might be better than `watch`. `watch` creates a subscription and might lead to performance issues